src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/PTMendRelationAction.java

```diff
@@ -0,0 +1,2348 @@
+            UndoRedoHandler.getInstance().add(result);
+            w1 = result.getNewWays().get(0);
+            wayToKeep = w1;
+        }
+        return wayToKeep;
+    }
+
+    private void findWayafterchunkRoundabout(Way way) {
+        List<Node> breakNode = new ArrayList<>();
+        breakNode.add(currentNode);
+        splitNode = way.lastNode();
+        Strategy strategy = new TempStrategyRoundabout();
+        List<List<Node>> wayChunks = SplitWayCommand.buildSplitChunks(way, breakNode);
+        SplitWayCommand result = SplitWayCommand.splitWay(way, wayChunks, Collections.emptyList(), strategy);
+        if (result != null) {
+            UndoRedoHandler.getInstance().add(result);
+        }
+    }
+
+    private void removeKeyListenerAndTemporaryLayer(KeyListener keyListener) {
+        MainApplication.getMap().mapView.removeKeyListener(keyListener);
+        MainApplication.getMap().mapView.removeTemporaryLayer(temporaryLayer);
+    }
+
+    void displayFixVariantsWithOverlappingWays(List<List<Way>> fixVariants) {
+        // find the letters of the fix variants:
+        char alphabet = 'A';
+        boolean numeric = PTAssistantPluginPreferences.NUMERICAL_OPTIONS.get();
+        wayListColoring = new HashMap<>();
+        final List<Character> allowedCharacters = new ArrayList<>();
+
+        if (numeric) {
+            alphabet = '1';
+            allowedCharacters.add('7');
+            if (showOption0)
+                allowedCharacters.add('0');
+            allowedCharacters.add('8');
+            allowedCharacters.add('9');
+        } else {
+            allowedCharacters.add('S');
+            if (showOption0)
+                allowedCharacters.add('W');
+            allowedCharacters.add('V');
+            allowedCharacters.add('Q');
+        }
+
+        for (int i = 0; i < 5 && i < fixVariants.size(); i++) {
+            allowedCharacters.add(alphabet);
+            wayListColoring.put(alphabet, fixVariants.get(i));
+            alphabet++;
+        }
+
+        // remove any existing temporary layer
+        removeTemporaryLayers();
+
+        if (abort)
+            return;
+
+        // zoom to problem:
+        AutoScaleAction.zoomTo(fixVariants.stream().flatMap(Collection::stream).collect(Collectors.toList()));
+
+        // display the fix variants:
+        temporaryLayer = new MendRelationAddMultipleLayer();
+        MainApplication.getMap().mapView.addTemporaryLayer(temporaryLayer);
+
+        // // add the key listener:
+        MainApplication.getMap().mapView.requestFocus();
+        MainApplication.getMap().mapView.addKeyListener(new KeyAdapter() {
+            @Override
+            public void keyPressed(KeyEvent e) {
+                downloadCounter = 0;
+                if (abort) {
+                    removeKeyListenerAndTemporaryLayer(this);
+                    return;
+                }
+                char typedKeyUpperCase = Character.toString(e.getKeyChar()).toUpperCase().toCharArray()[0];
```

Comment:
I'd suggest creating an enum with all possible actions, then here instead of converting to uppercase, convert to that enum. Then later decide based on that enum what to do. Then all the case differentiations if we are in numeric mode or ABC mode won't be needed anymore.
Also e.g. the colors could be selected by simply doing `FIVE_COLOR_PALETTE[enumValue.ordinal()]`.